http/src/request/channel/reaction/mod.rs

```diff
@@ -29,19 +73,99 @@ impl From<ReactionType> for RequestReactionType {
     }
 }
 
-fn format_emoji(emoji: RequestReactionType) -> String {
-    match emoji {
-        RequestReactionType::Custom { id, name } => {
-            let mut emoji = String::new();
-            match name {
-                Some(name) => emoji.push_str(name.as_ref()),
-                None => emoji.push('e'),
+/// Format a [`RequestReactionType`] into a format acceptable for use in URLs.
+///
+/// # Examples
+///
+/// Format a custom reaction for use in a URL:
+///
+/// ```
+/// use twilight_http::request::channel::reaction::RequestReactionType;
+/// use twilight_model::id::EmojiId;
+///
+/// let reaction = RequestReactionType::Custom {
+///     id: EmojiId(123),
+///     name: Some("rarity".to_owned()),
+/// };
+///
+/// // Retrieve the display formatter.
+/// let display = reaction.display();
+///
+/// // And now format it into an acceptable string and then check it.
+/// assert_eq!("rarity:123", display.to_string());
+/// ```
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct RequestReactionTypeDisplay<'a>(&'a RequestReactionType);
+
+impl Display for RequestReactionTypeDisplay<'_> {
+    fn fmt(&self, f: &mut Formatter<'_>) -> FmtResult {
+        match self.0 {
+            RequestReactionType::Custom { id, name } => {
+                if let Some(name) = name {
+                    f.write_str(name)?;
+                } else {
+                    f.write_str("e")?;
+                }
+
+                f.write_str(":")?;
+
+                Display::fmt(id, f)
+            }
+            RequestReactionType::Unicode { name } => {
+                Display::fmt(&utf8_percent_encode(&name, NON_ALPHANUMERIC), f)
             }
-            let _ = write!(emoji, ":{}", id);
-            emoji
-        }
-        RequestReactionType::Unicode { name } => {
-            utf8_percent_encode(&name, NON_ALPHANUMERIC).to_string()
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
+    // `clippy::non_ascii_literal` can't be allowed on an item level; it can
+    // only be enabled on a module level.
+    #![allow(clippy::non_ascii_literal)]
```

Comment:
What is this being used for?

Reply:
for the emojis that would otherwise need to be written like `\Uxxxx`

Reply:
Til